pipeline_dp/dp_engine.py

```diff
@@ -185,6 +192,7 @@ def filter_fn(captures: Tuple[MechanismSpec, int],
 
         # make filter_fn serializable
         filter_fn = partial(filter_fn, (budget, max_partitions_contributed))
+        self._add_report_stage(f"Select private partitions")
```

Comment:
Maybe to have format for all messages like that:
<name of the stage>: <some additional information> (for contribution bounding it's the case).

Here it might be like
**Private Partition selection: using Truncated Geometric method with (eps=..., delta = )**

WDYT?

Reply:
`eps, delta` can be taken from `budget`. But eps, delta are evaluated lazily, so which is not a problem, you can pass lambda returning to `_add_report_stage`, i.e.
_add_report_stage(lambda : f".... {budget.eps} ...")

Reply:
Looks great, I've implemented this lazily 😎